lib/omniauth/strategies/saml.rb

```diff
@@ -69,7 +69,7 @@ def response_fingerprint
       end
 
       def other_phase
-        if current_path.start_with?(request_path)
+        if current_path.match(/#{Regexp.quote(request_path)}[\/\z]{1}/)
```

Comment:
There are a few problems with this regexp:

1. It needs to be anchored at the beginning. It currently can math things that have a prefix in front of `request_path`
2. I'm pretty sure `\z` doesn't work inside a character class. You need to use `(/|\z)`
3. The `{1}` isn't doing anything.

Since the underlying implementation of `request_path` memoizes its value and doesn't change over the lifecycle of the strategy instance, I think I would introduce a method like this:

```ruby
def request_path_pattern
  @request_path_pattern ||= /\A#{Regexp.quote(request_path)}(/|\z)/
end
```

Then, in `other_phase`, I would call `if request_path_pattern.match(current_path)`.

Reply:
This will still break if someone uses `options[:request_path]` and puts a trailing slash on it, but it's better than the current situation.